contracts-periphery/script/Deploy.s.sol

```diff
@@ -1,15 +1,29 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.16;
 
-import "forge-std/Script.sol";
+import {Script} from "forge-std/Script.sol";
+import {StdChains} from "forge-std/StdChains.sol";
 import {UmbraBatchSend} from "src/UmbraBatchSend.sol";
 import {IUmbra} from "src/interface/IUmbra.sol";
 
 contract Deploy is Script {
   address constant UMBRA = 0xFb2dc580Eed955B528407b4d36FfaFe3da685401;
 
-  function run() public {
+  function deploy() internal {
     vm.broadcast();
     new UmbraBatchSend(IUmbra(UMBRA));
   }
 }
+
+contract ScriptBase is Deploy {
+  string[] public networks = ["mainnet", "optimism", "arbitrum_one", "polygon", "goerli", "sepolia"];
+
+  function run(address expectedContractAddress) public {
```

Comment:
Can we compute the `expectedContractAddress` in the script instead? Feels a lot safer that way. We'll know the deployer address, we'll know the expected nonce, and we can use `computeCreateAddress` from forge-std. This also simplifies the deploy command by removing the need for the `--sig` arg since `run()` is the default sig it looks for

Reply:
I knew this must be possible but couldn't find 'computeCreateAddress' easily from the Foundrybook. Maybe when I get the feature where chatGPT can crawl URLs that you provide, I would be able to find it and find it faster.

Reply:
yea it's part of forge-std, not foundry, so might not be in the book: https://github.com/foundry-rs/forge-std/blob/73d44ec7d124e3831bc5f832267889ffb6f9bc3f/src/StdUtils.sol#L93

Reply:
It's actually in the book: https://book.getfoundry.sh/reference/forge-std/compute-create-address?highlight=computeCreateAddress#computecreateaddress

Implemented changes here: https://github.com/ScopeLift/umbra-protocol/pull/504/commits/a6862bcf242ee41664cb58fd0810cba3bc65003e